sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -502,6 +502,30 @@ func (ihnc *indexHashedNodesCoordinator) GetAllLeavingValidatorsPublicKeys(epoch
 	return validatorsPubKeys, nil
 }
 
+// GetAllShuffledOutValidatorsPublicKeys will return all shuffled out validator public keys from all shards
+func (ihnc *indexHashedNodesCoordinator) GetAllShuffledOutValidatorsPublicKeys(epoch uint32) (map[uint32][][]byte, error) {
+	validatorsPubKeys := make(map[uint32][][]byte)
+
+	ihnc.mutNodesConfig.RLock()
+	nodesConfig, ok := ihnc.nodesConfig[epoch]
+	ihnc.mutNodesConfig.RUnlock()
+
+	if !ok {
+		return nil, fmt.Errorf("%w epoch=%v", ErrEpochNodesConfigDoesNotExist, epoch)
+	}
+
+	nodesConfig.mutNodesMaps.RLock()
+	defer nodesConfig.mutNodesMaps.RUnlock()
+
+	for shardID, shuffledOutList := range nodesConfig.shuffledOutMap {
```

Comment:
it is a very small and extreme edge case. If, at first RLock you'll read the epoch 70 and `ok` will be `true` but until your next RLock it will be removed from the map, then you will range over an empty map, and instead of returning an error, you will return an empty map.
I would suggest that you extend the RLock - RUnlock area